src/L2DebtManager.sol

```diff
@@ -1,262 +1,502 @@
+     */
+    function liquidWeEthAmount() public view returns (uint256) {
+        return
+            weETH.balanceOf(address(this)) -
+            _totalCollateralAmounts[address(weETH)];
     }
 
-    function borrowingOf(address user) public view returns (uint256) {
-        return _borrowings[user];
+    /**
+     * @inheritdoc IL2DebtManager
+     */
+    function liquidUsdcAmount() public view returns (uint256) {
+        return usdc.balanceOf(address(this)) - _totalBorrowingAmount;
     }
 
-    /// Debt ratio is calculated as the ratio of the debt to the collateral value in USDC
-    // it returns the ratio in basis points (1e4)
-    function debtRatioOf(address user) public view returns (uint256) {
-        uint256 eEthPriceInUSDC = getEEthPriceInUsdc(); // assuming this returns the price scaled to 6 decimals
-        uint256 debtValue = _borrowings[user];
-        uint256 collateralValue = (_collaterals[user] * eEthPriceInUSDC) / 1e18; // adjust for eETH's 18 decimals
-
-        require(collateralValue > 0, "ZERO_COLLATERAL_VALUE");
+    /**
+     * @inheritdoc IL2DebtManager
+     */
+    function convertUsdcToCollateralToken(
+        address collateralToken,
+        uint256 debtUsdcAmount
+    ) public view returns (uint256) {
+        if (collateralToken != address(weETH))
+            revert UnsupportedCollateralToken();
+        return (debtUsdcAmount * 1e18) / priceProvider.getWeEthUsdPrice();
+    }
 
-        return (debtValue * 1e4) / collateralValue; // result in basis points
+    /**
+     * @inheritdoc IL2DebtManager
+     */
+    function convertCollateralTokenToUsdc(
+        address collateralToken,
+        uint256 collateralAmount
+    ) public view returns (uint256) {
+        if (collateralToken != address(weETH))
+            revert UnsupportedCollateralToken();
+        return (collateralAmount * priceProvider.getWeEthUsdPrice()) / 1e18;
     }
 
-    function remainingBorrowingCapacityInUSDC(
+    /**
+     * @inheritdoc IL2DebtManager
+     */
+    function getCollateralValueInUsdc(
         address user
     ) public view returns (uint256) {
-        uint256 eEthPriceInUSDC = getEEthPriceInUsdc();
-        uint256 collateralValue = (_collaterals[user] * eEthPriceInUSDC) / 1e18;
-        uint256 maxBorrowingAmount = (collateralValue * liquidationThreshold) /
-            1e4;
         return
-            maxBorrowingAmount > _borrowings[user]
-                ? maxBorrowingAmount - _borrowings[user]
-                : 0;
+            (_userCollateral[user][address(weETH)] *
+                priceProvider.getWeEthUsdPrice()) / 1e18;
     }
 
-    function liquidEEthAmount() public view returns (uint256) {
-        return eETH.balanceOf(address(this)) - totalCollateralAmount;
+    /**
+     * @inheritdoc IL2DebtManager
+     */
+    function collateralTokens()
+        external
+        view
+        returns (address[] memory tokens)
+    {
+        tokens = new address[](1);
+        tokens[0] = address(weETH);
```

Comment:
shuold the contract maintain a list of collateral tokens instead of hard-coding it at multiple places?

Reply:
I think we can do it that way as well. I will make that change.